dags/als-kp-dataset-dag.py

```diff
@@ -161,6 +161,117 @@ def transform_with_jsonata(
     return transformed_items, validation_errors
 
 
+
+def parse_dataset_code(dataset_code: str) -> Tuple[str, str, str]:
+    """Parse dataset code to extract ALS number, prefix, and date.
+
+    Args:
+        dataset_code: e.g., "src_als1003_2025_04_17", "fv1_als1001_2025_02_26"
+
+    Returns:
+        Tuple of (als_number, prefix, date_str)
+
+    Raises:
+        ValueError: If dataset code format is invalid
+    """
+    import re
+
+    # Pattern to match: prefix_als{number}_{date}
+    pattern = r'^(src_|fm[12]_|fv[123]_)(als\d+)_(\d{4}_\d{2}_\d{2})$'
+    match = re.match(pattern, dataset_code)
+
+    if not match:
+        raise ValueError(f"Invalid dataset code format: {dataset_code}")
+
+    prefix = match.group(1).rstrip('_')  # Remove trailing underscore
+    als_number = match.group(2)
+    date_str = match.group(3)
+
+    return als_number, prefix, date_str
+
+
+def extract_dataset_code_from_url(url: str) -> str:
+    """Extract dataset code from C-Path URL.
+
+    Args:
+        url: e.g., "https://fair.dap.c-path.org/#/data/datasets/src_als1003_2025_04_17"
+
+    Returns:
+        Dataset code e.g., "src_als1003_2025_04_17"
+    """
+    if not url:
+        return ""
+    return url.split("/")[-1]
+
+
+def get_version_priority(prefix: str) -> int:
+    """Get priority score for dataset prefix (higher = better).
+
+    Priority: fv3 > fv2 > fv1 > fm2 > fm1 > src
+    """
+    priority_map = {
+        'fv3': 6,
+        'fv2': 5,
+        'fv1': 4,
+        'fm2': 3,
+        'fm1': 2,
+        'src': 1
+    }
+    return priority_map.get(prefix, 0)
+
+
+def select_latest_versions(datasets: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    """Select the latest and greatest version for each ALS dataset group.
+
+    Args:
+        datasets: List of dataset items with 'url' field containing dataset codes
+
+    Returns:
+        List of selected datasets (one per ALS number)
+    """
+    from collections import defaultdict
+    from datetime import datetime
+
+    # Group datasets by ALS number
+    als_groups = defaultdict(list)
+
+    for dataset in datasets:
+        url = dataset.get("url", "")
+        dataset_code = extract_dataset_code_from_url(url)
+
+        if not dataset_code:
```

Comment:
Datasets without a `url` will be silently skipped. Is that the expected behavior?